src/test/java/teammates/test/cases/webapi/SessionLinksRecoveryActionTest.java

```diff
@@ -0,0 +1,311 @@
+    private FeedbackSessionAttributes session2InCourse1 = typicalBundle.feedbackSessions.get("session2InCourse1");
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.SESSION_LINKS_RECOVERY;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @BeforeMethod
+    @Override
+    public void beforeTestMethodSetup() {
+        // change start time of one feedback session in the data bundle for the purpose of this testing.
+        DataBundle dataBundle = typicalBundle;
+
+        // opened and unpublished.
+        session1InCourse3.setStartTime(TimeHelper.getInstantDaysOffsetFromNow(-20));
+        dataBundle.feedbackSessions.put("session1InCourse3", session1InCourse3);
+
+        // closed and unpublished
+        session2InCourse3.setStartTime(TimeHelper.getInstantDaysOffsetFromNow(-19));
+        session2InCourse3.setEndTime(TimeHelper.getInstantDaysOffsetFromNow(-1));
+        session2InCourse3.resetDeletedTime();
+        dataBundle.feedbackSessions.put("session2InCourse3", session2InCourse3);
+
+        // opened and published.
+        session1InCourse1.setStartTime(TimeHelper.getInstantDaysOffsetFromNow(-19));
+        session1InCourse1.setResultsVisibleFromTime(TimeHelper.getInstantDaysOffsetFromNow(-1));
+        dataBundle.feedbackSessions.put("session1InCourse1", session1InCourse1);
+
+        // closed and published
+        session2InCourse1 = dataBundle.feedbackSessions.get("session2InCourse1");
+        session2InCourse1.setStartTime(TimeHelper.getInstantDaysOffsetFromNow(-18));
+        session2InCourse1.setEndTime(TimeHelper.getInstantDaysOffsetFromNow(-1));
+        session2InCourse1.setResultsVisibleFromTime(TimeHelper.getInstantDaysOffsetFromNow(-1));
+        dataBundle.feedbackSessions.put("session2InCourse1", session2InCourse1);
+
+        removeAndRestoreDataBundle(dataBundle);
+    }
+
+    @Test
+    protected void testExecute_notEnoughParameters() {
+        ______TS("Invalid parameters");
+        // no params
+        verifyHttpParameterFailure();
+    }
+
+    @Test
+    protected void testExecute_nonExistingEmail() {
+        ______TS("Typical case: non-existing email address");
+
+        String[] nonExistingParam = new String[] {
+                Const.ParamsNames.SESSION_LINKS_RECOVERY_EMAIL, "non-existent email address",
+        };
+
+        SessionLinksRecoveryAction a = getAction(nonExistingParam);
+        JsonResult result = getJsonResult(a);
+
+        SessionLinksRecoveryResponseData output = (SessionLinksRecoveryResponseData) result.getOutput();
+
+        assertEquals("The recovery links for your feedback sessions have been sent to "
+                + "the specified email address: non-existent email address", output.getMessage());
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+        verifyNumberOfEmailsSent(a, 1);
```

Comment:
it doesnt make sense since this is just a random string, not even a valid email address. Can we check whether the param input is a valid email address?

Reply:
I would suggest we check it in frontend before submission? @amrut-prabhu  What do you think?

Reply:
uhmm frontend validations r definitely not complete. If i directly send request to backend e.g. using postman with invalid param values, unnecessary errors will be generated.

Reply:
I do not think validate email format at backend is a good idea as it does not provide good user experience(it has to wait until user click submit then he knows that the input is not even valid.)

Suggestion will be either we handle it in frontend. BTW invalid email will not cause any errors, it will just tell frontend the email fail to send (through the flag)

Reply:
@junming403 back-end validation in absolutely necessary because the system's stability depends on it, front-end validation is necessary only if it enhances user experience. This applies to 99% of other occurrences in the world, not just here, not even just in TEAMMATES.

Reply:
Understood, thanks for pointing this out.